hera_sim/visibilities/fftvis.py

```diff
@@ -0,0 +1,318 @@
+            + nsrc * nf  # source fluxes
+            + nt * 9  # rotation matrices
+            + 3 * nsrc
+            + 3 * nsrc  # source positions (topo and eq)
+        )
+
+        return all_floats * self._precision * 4 / 1024**3
+
+    def simulate(self, data_model):
+        """
+        Calls :func:fftvis to perform the visibility calculation.
+
+        Returns
+        -------
+        array_like of self._complex_dtype
+            Visibilities. Shape=self.uvdata.data_array.shape.
+        """
+        polarized = self._check_if_polarized(data_model)
+        feed = self.get_feed(data_model.uvdata)
+
+        # Setup MPI info if enabled
+        if self.mpi_comm is not None:
+            myid = self.mpi_comm.Get_rank()
+            nproc = self.mpi_comm.Get_size()
+
+        if self.correct_source_positions:
+            ra, dec = self.correct_point_source_pos(data_model)
+            logger.info("Done correcting source positions.")
+        else:
+            ra, dec = data_model.sky_model.ra, data_model.sky_model.dec
+
+        logger.info("Getting Equatorial Coordinates")
+        crd_eq = convs.point_source_crd_eq(ra, dec)
+
+        # Convert equatorial to topocentric coords
+        logger.info("Getting Rotation Matrices")
+        eq2tops = self.get_eq2tops(data_model.uvdata, data_model.lsts)
+
+        # The following are antenna positions in the order that they are
+        # in the uvdata.data_array
+        active_antpos_array, ant_list = data_model.uvdata.get_ENU_antpos(
+            pick_data_ants=True
+        )
+        active_antpos = {
+            ant_index: antpos
+            for ant_index, antpos in zip(ant_list, active_antpos_array)
+        }
+
+        # Get pixelized beams if required
+        logger.info("Preparing Beams...")
+        beam = convs.prepare_beam(
+            data_model.beams[0],
+            polarized=polarized,
+            use_feed=feed,
+        )
+
+        # Get all the polarizations required to be simulated.
+        req_pols = self._get_req_pols(
+            data_model.uvdata, data_model.beams[0], polarized=polarized
+        )
+
+        # Empty visibility array
+        if np.all(data_model.uvdata.data_array == 0):
+            # Here, we don't make new memory, because that is just a whole extra copy
+            # of the largest array in the calculation. Instead we fill the data_array
+            # directly.
+            visfull = data_model.uvdata.data_array
+        else:
+            visfull = np.zeros_like(
+                data_model.uvdata.data_array, dtype=self._complex_dtype
+            )
+
+        for i, freq in enumerate(data_model.freqs):
+            # Divide tasks between MPI workers if needed
+            if self.mpi_comm is not None and i % nproc != myid:
+                continue
+
+            logger.info(f"Simulating Frequency {i+1}/{len(data_model.freqs)}")
+
+            # Call fftvis function to simulate visibilities
+            vis = self._fftvis(
```

Comment:
Repeating the above comment -- let's just use the `fftvis.simulate.simulate` directly here.

Reply:
Fixed this